### PR TITLE
BGDIINF_SB-2445: Python3 support bug fix

### PR DESCRIPTION
**Pull request by ltshb**
Python3 doesn't have urllib2 but urllib.

**TO BE MERGED AFTER #3924**

### CODE REVIEW

**Comment by ltclm** [comment]
thank you @ltshb 
https://jira.swisstopo.ch/browse/BGDIINF_SB-2445 can be closed

**Review by hansmannj — APPROVED** [review]
:1st_place_medal:

**Comment by rebert** [comment]
@ltshb Thanks for your work. I'm pretty sure that the kgs mako does not work yet. But this is more related to the csv reader ...

File: chsdi/templates/htmlpopup/ivs_nat.mako [hunk 1]
```diff
@@ -30,7 +30,11 @@
     <tr><td class="cell-left">${_(ivs_slaname)}</td><td>${c['attributes']['ivs_slaname']}</td></tr>
     <tr><td class="cell-left">${_('ivs_documentation')}</td>
 <%
-    from urllib2 import urlopen
```
Comment (rebert) [inline chsdi/templates/htmlpopup/ivs_nat.mako:33]: it makes no sense just to import libraries without using them. I would just remove the import

**Review by rebert — CHANGES_REQUESTED** [review]

File: chsdi/templates/htmlpopup/kgs.mako [hunk 2]
```diff
@@ -17,7 +17,11 @@
         objarts = c['attributes']['objektart'].split(',')
         gemeinde_ehemalig = c['attributes']['gemeinde_ehemalig'] if c['attributes']['gemeinde_ehemalig'].startswith("(") else "("+c['attributes']['gemeinde_ehemalig']+")"
         import csv
-        from urllib2 import urlopen
+        try:
+            from urllib2 import urlopen
+        except ImportError:
+            # Python3 fallback
+            from urllib.request import urlopen
```
Comment (rebert) [inline chsdi/templates/htmlpopup/kgs.mako:24]: The make does not work in py3 yet. 
/rest/services/ech/MapServer/ch.babs.kulturgueter/9980/extendedHtmlPopup?lang=en
``pyramid_mako.MakoRenderingException: 

Traceback (most recent call last):
  File "/home/ltrea/mf-chsdi3/.venv/lib/python3.7/site-packages/pyramid_mako/__init__.py", line 148, in __call__
    result = template.render_unicode(**system)
  File "/home/ltrea/mf-chsdi3/.venv/lib/python3.7/site-packages/mako/template.py", line 482, in render_unicode
    self, self.callable_, args, data, as_unicode=True
  File "/home/ltrea/mf-chsdi3/.venv/lib/python3.7/site-packages/mako/runtime.py", line 883, in _render
    **_kwargs_for_callable(callable_, data)
  File "/home/ltrea/mf-chsdi3/.venv/lib/python3.7/site-packages/mako/runtime.py", line 920, in _render_context
    _exec_template(inherit, lclcontext, args=args, kwargs=kwargs)
  File "/home/ltrea/mf-chsdi3/.venv/lib/python3.7/site-packages/mako/runtime.py", line 947, in _exec_template
    callable_(context, *args, **kwargs)
  File "/home/ltrea/mf-chsdi3/chsdi/templates/htmlpopup/base.mako", line 52, in render_body
    ${self.extended_info(c, lang)}
  File "/home/ltrea/mf-chsdi3/chsdi/templates/htmlpopup/kgs.mako", line 31, in render_extended_info
    next(reader) # Skip header
Error: iterator should return strings, not bytes (did you open the file in text mode?```

**Review by rebert — CHANGES_REQUESTED** [review]

**Comment by ltclm** [comment]
> @ltshb Thanks for your work. I'm pretty sure that the kgs mako does not work yet (in py3). But this is more related to the csv reader ...
> 
> f.ex http://localhost:9081/rest/services/ech/MapServer/ch.babs.kulturgueter/9980/extendedHtmlPopup?lang=en

the csv reader is failing because the data.geo.admin.ch url is not valid. we should use data.geo.admin.ch instead, we dont have any dev or int environment for data.geo.admin.ch

image

File: chsdi/templates/htmlpopup/kgs.mako (hunk 2 above)
Comment (ltshb) [reply chsdi/templates/htmlpopup/kgs.mako:24]: @rebert but this has nothing to do with this PR. We can already merge this and fix the other python3 issues later on.

**Comment by rebert** [comment]
@ltshb shall I make a extra ticket for this tooltip?

**Review by rebert — APPROVED** [review]
urllib2 has been replaced. this lgtm. But the tooltip kgs does not work yet. For several reasons

**Comment by ltshb** [comment]
> @ltshb shall I make a extra ticket for this tooltip?

I would first deploy this on k8s and test it again on k8s and if still an issue yes.

**Pull request merged**